File: 387__First_Unique_Character_in_a_String.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        dic = {}
        for char in s:
            if char in dic:
                dic[char] += 1
            else:
                dic[char] = 1

        for i, char in enumerate(s):
            if dic[char] == 1:
                return i


        return -1

File: Java/387__First_Unique_Character_in_a_String.java
class Solution {
    public int firstUniqChar(String s) {
        HashMap<Character, Integer> map = new HashMap<>();
        char[] lst = s.toCharArray();
        for (int i = 0; i < lst.length; i++) {
            map.put(lst[i], map.getOrDefault(lst[i], 0) + 1);
        }
        for (int i = 0; i < lst.length; i++) {
            if (map.get(lst[i]) == 1) {
                return i;
            }
        }
        return -1;
    }
}
